addons/website_slides/wizard/slide_channel_invite.py

```diff
@@ -19,34 +19,50 @@ class SlideChannelInvite(models.TransientModel):
 
     # composer content
     attachment_ids = fields.Many2many('ir.attachment', string='Attachments')
+    send_email = fields.Boolean('Send Email')
     # recipients
     partner_ids = fields.Many2many('res.partner', string='Recipients')
     # slide channel
-    channel_id = fields.Many2one('slide.channel', string='Slide channel', required=True)
+    channel_id = fields.Many2one('slide.channel', string='Course', required=True)
+    channel_invite_url = fields.Char('Course Link', compute='_compute_channel_invite_url')
+    channel_visibility = fields.Selection(related='channel_id.visibility')
+    channel_published = fields.Boolean(related='channel_id.is_published')
+    # membership
+    with_enrollment = fields.Boolean(
+        'Enroll partners', readonly=True,
+        help='Whether invited partners will be added as enrolled. Otherwise, they will be added as invited.')
+
+    @api.depends('channel_id')
+    def _compute_channel_invite_url(self):
+        for invite in self:
+            channel = invite.channel_id
+            invite.channel_invite_url = f'{channel.get_base_url()}/slides/{channel.id}/invite'
+
+    @api.onchange('channel_visibility')
+    def _onchange_channel_visibility(self):
```

Comment:
Why not making "send_email" a stored computed field based on channel_id and with_enrollment ? That way we avoid an onchange.